inc/woocommerce.php

```diff
@@ -0,0 +1,239 @@
+	}
+}
+
+add_action( 'woocommerce_after_shop_loop', '_s_woocommerce_product_columns_wrapper_close', 40 );
+
+/**
+ * Remove default WooCommerce wrapper
+ */
+remove_action( 'woocommerce_before_main_content', 'woocommerce_output_content_wrapper', 10 );
+remove_action( 'woocommerce_after_main_content', 'woocommerce_output_content_wrapper_end', 10 );
+
+if ( ! function_exists( '_s_woocommerce_wrapper_before' ) ) {
+	/**
+	 * Before Content
+	 * Wraps all WooCommerce content in wrappers which match the theme markup
+	 *
+	 * @return void
+	 */
+	function _s_woocommerce_wrapper_before() {
+		?>
+		<div id="primary" class="content-area">
+			<main id="main" class="site-main" role="main">
+		<?php
+	}
+}
+
+add_action( 'woocommerce_before_main_content', '_s_woocommerce_wrapper_before' );
+
+if ( ! function_exists( '_s_woocommerce_wrapper_after' ) ) {
+	/**
+	 * After Content
+	 * Closes the wrapping divs
+	 *
+	 * @return void
+	 */
+	function _s_woocommerce_wrapper_after() {
+		?>
+			</main><!-- #main -->
+		</div><!-- #primary -->
+		<?php
+	}
+}
+
+add_action( 'woocommerce_after_main_content', '_s_woocommerce_wrapper_after' );
+
+if ( ! function_exists( '_s_woocommerce_cart_link_fragment' ) ) {
+	/**
+	 * Cart Fragments
+	 * Ensure cart contents update when products are added to the cart via AJAX
+	 *
+	 * @param array $fragments Fragments to refresh via AJAX.
+	 * @return array Fragments to refresh via AJAX
+	 */
+	function _s_woocommerce_cart_link_fragment( $fragments ) {
+		global $woocommerce;
+
+		ob_start();
+		_s_woocommerce_cart_link();
+		$fragments['a.cart-contents'] = ob_get_clean();
+
+		return $fragments;
+	}
+}
+
+add_filter( 'woocommerce_add_to_cart_fragments', '_s_woocommerce_cart_link_fragment' );
+
+if ( ! function_exists( '_s_woocommerce_cart_link' ) ) {
+	/**
+	 * Cart Link
+	 * Displayed a link to the cart including the number of items present and the cart total
+	 *
+	 * @return void
+	 */
+	function _s_woocommerce_cart_link() {
+		?>
+			<a class="cart-contents" href="<?php echo esc_url( wc_get_cart_url() ); ?>" title="<?php esc_attr_e( 'View your shopping cart', '_s' ); ?>">
+				<?php /* translators: number of items in the mini cart. */ ?>
+				<span class="amount"><?php echo wp_kses_data( WC()->cart->get_cart_subtotal() ); ?></span> <span class="count"><?php echo wp_kses_data( sprintf( _n( '%d item', '%d items', WC()->cart->get_cart_contents_count(), '_s' ), WC()->cart->get_cart_contents_count() ) );?></span>
+			</a>
+		<?php
+	}
+}
+
+if ( ! function_exists( '_s_woocommerce_header_cart' ) ) {
+	/**
+	 * Display Header Cart
+	 *
+	 * @return void
+	 */
+	function _s_woocommerce_header_cart() {
+		if ( is_cart() ) {
+			$class = 'current-menu-item';
+		} else {
+			$class = '';
+		}
+		?>
+		<ul id="site-header-cart" class="site-header-cart">
+			<li class="<?php echo esc_attr( $class ); ?>">
+				<?php _s_woocommerce_cart_link(); ?>
+			</li>
+			<li>
+				<?php the_widget( 'WC_Widget_Cart', 'title=' ); ?>
```

Comment:
According to the documentation the second argument is the instance not the arguments. https://developer.wordpress.org/reference/functions/the_widget/

It would be better to create an array of arguments instead.